### ISSUE TITLE
consider underlines for links

### ISSUE THREAD

**Issue by meganrogge** [microsoft/vscode#203989]
We currently distinguish links using only color, no underline in some places.

This can make it hard for color blind or low vision users to differentiate between links and regular text. 

https://aka.ms/MAS1.4.1

Underlines can also reduce readability though. 

Thoughts @hbons or @isidorn ?

https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1881734

![Screenshot 2024-02-01 at 11 07 51 AM](https://github.com/microsoft/vscode/assets/29464607/54c68f87-116c-4c67-9749-1cd3d0c2e69e)
![Screenshot 2024-02-01 at 11 07 59 AM](https://github.com/microsoft/vscode/assets/29464607/b3e8cabf-a1c2-482d-81b9-9003674630ff)

**Comment by setaskin** [comment]
Hi all, Live Share has a sev2 bug assigned for it. It effects our Accessibility Grade. I filed an extension request for it.

**Comment by hbons** [comment]
How about we start with enabling this in the High Contrast themes?

**Comment by hbons** [comment]
This one is trickier than I thought. VS Code uses `<a>` elements throughout the IDE also for things like tabs and buttons, so we need to find a way to only match the elements that look like text...

**Comment by meganrogge** [comment]
I noticed that we do this in the editor even for non-HC themes.
![Screenshot 2024-02-05 at 12 32 10 PM](https://github.com/microsoft/vscode/assets/29464607/e38c2792-2a68-4670-9a90-94d2a3e358e0)

**Comment by hbons** [comment]
After some investigation, matching `[role=link]` doesn't work as it's the default role and we can only match it if it's explicitly set in the `a` element. Looking for some other solutions, but we may need to add the role attribute explicitly everywhere.

**Comment by hbons** [comment]
@isidorn do you have advice how we might work around this?

**Comment by isidorn** [comment]
We do not have an issue filled for this as is my understanding.
LiveShare has it, and filling extension is the right approach (which was already done).

So for now I suggest no action here.

**Comment by hbons** [comment]
I still think this is a good thing to look into. Our links in the HC themes don't look very contrasty in any case:

<img width="722" alt="image" src="https://github.com/microsoft/vscode/assets/103326/005b7f04-f449-473d-bf19-be012517b46d">

I realised we don't need to underline all links, just those that appear in a text. So matching `p a` should do the trick.

**Comment by isidorn** [comment]
@hbons that's fair. So could we fix for High Contast only first?

**Comment by hbons** [comment]
@isidorn yes, we'll keep it to HC themes for now. I've managed to get it to match only relevant text links that have surrounding text. As you can see standalone text links won't be underlined. GitHub also does not do this.

I think it looks pretty good (ignore the red colour, it's to check we're matching the right things and I will reset it):

<img width="1150" alt="image" src="https://github.com/microsoft/vscode/assets/103326/9c06f8e1-1f7a-4e08-8b9e-dc8f235a736d">

<img width="421" alt="image" src="https://github.com/microsoft/vscode/assets/103326/eeff0a9e-eee1-4a72-9478-f9d1de405d18">

<img width="943" alt="image" src="https://github.com/microsoft/vscode/assets/103326/86bf888f-a32b-43b2-9d35-5897f5e51747">

**Comment by hbons** [comment]
Specially in the light HC theme it's big improvement:
<img width="343" alt="image" src="https://github.com/microsoft/vscode/assets/103326/22b55733-6d0a-4182-b706-2df7ac3a37b4">

**Comment by isidorn** [comment]
Looking good for the HC theme. Thank you

**Comment by alexr00** [comment]
Should these have underlines in HC?
![image](https://github.com/microsoft/vscode/assets/38270282/538d2dc7-7aa6-44b6-9b89-edece7b95e67)

Otherwise, verified in welcome views and chat.

**Comment by hbons** [comment]
@alexr00 these don't need to be underlined. Only links in a paragraph surrounded by other text.

**Comment by daviddossett** [comment]
Adding setting to enable link underlines in https://github.com/microsoft/vscode/pull/216842
